--- a/backend/internal/certificate.js
+++ b/backend/internal/certificate.js
@@ -20,7 +20,7 @@
 import internalNginx from "./nginx.js";
 
 const letsencryptConfig = "/etc/letsencrypt.ini";
-const certbotCommand = "certbot";
+const certbotDefaultCommand = "certbot";
 const certbotLogsDir = "/data/logs";
 const certbotWorkDir = "/tmp/letsencrypt-lib";
 
@@ -779,6 +779,8 @@
 			`Requesting LetsEncrypt certificates for Cert #${certificate.id}: ${certificate.domain_names.join(", ")}`,
 		);
 
+		const certbotCommand = certbotDefaultCommand;
+
 		const args = [
 			"certonly",
 			"-n", // non-interactive
@@ -835,6 +837,8 @@
 		// Whether the plugin has a --<name>-credentials argument
 		const hasConfigArg = certificate.meta.dns_provider !== "route53";
 
+		const certbotCommand = `${certbotDefaultCommand}-${dnsPlugin.full_plugin_name}`;
+
 		const args = [
 			"certonly",
 			"-n", // non-interactive
@@ -945,6 +949,8 @@
 			`Renewing LetsEncrypt certificates for Cert #${certificate.id}: ${certificate.domain_names.join(", ")}`,
 		);
 
+		const certbotCommand = certbotDefaultCommand;
+
 		const args = [
 			"renew",
 			"--force-renewal",
@@ -1028,6 +1034,8 @@
 			);
 		}
 
+		const certbotCommand = `${certbotDefaultCommand}-${dnsPlugin.full_plugin_name}`;
+
 		const args = [
 			"renew",
 			"--force-renewal",
@@ -1076,6 +1084,12 @@
 	 * @returns {Promise}
 	 */
 	revokeLetsEncryptSsl: async (certificate, throwErrors) => {
+		let certbotCommand = certbotDefaultCommand;
+		if (certificate.meta.dns_challenge) {
+			const dnsPlugin = dnsPlugins[certificate.meta.dns_provider];
+			certbotCommand = `${certbotCommand}-${dnsPlugin.full_plugin_name}`;
+		}
+
 		logger.info(
 			`Revoking LetsEncrypt certificates for Cert #${certificate.id}: ${certificate.domain_names.join(", ")}`,
 		);
