--- a/backend/lib/certbot.js
+++ b/backend/lib/certbot.js
@@ -70,7 +70,7 @@
 			env = Object.assign(env, plugin.env);
 		}
 
-		const cmd = `. /opt/certbot/bin/activate && pip install --no-cache-dir ${plugin.dependencies} ${plugin.package_name}${plugin.version}  && deactivate`;
+		const cmd = 'pip install --no-cache-dir ' + plugin.dependencies + ' ' + plugin.package_name + plugin.version;
 		return utils.exec(cmd, {env})
 			.then((result) => {
 				logger.complete(`Installed ${pluginKey}`);
